### ISSUE TITLE
Bug: there is no error about forbidden vertical imports

### ISSUE THREAD

**Issue by EliseyMartynov** [feature-sliced/steiger#49]
May be I'm doing something wrong, but I can't get an error about forbidden import hierarchy, e.g. when importing `app` in `entities`.
Here is the sandbox: https://codesandbox.io/p/devbox/steiger-test-g4f4wp

**Comment by illright** [comment]
Indeed, seems like a bug. Thanks for reporting!

**Comment by daniilsapa** [comment]
@illright 
Could you assign this task to me?

**Comment by daniilsapa** [comment]
As I see the forbidden-imports rule works just as expected. Except for a single case when you try to import for an index file of the layer because it thinks that a layer cannot have an index file

<img width="864" alt="Screenshot 2024-07-18 at 15 19 02" src="https://github.com/user-attachments/assets/31825231-8f3d-4f2b-a2a4-5b9a49116915">

E.g. I created a `/ui` folder inside `app` and then tried to import it into the cart entity, it reported errors just fine.

<img width="863" alt="Screenshot 2024-07-18 at 15 17 52" src="https://github.com/user-attachments/assets/8b688fe5-a757-4110-9005-eff0452103e0">

Now when we added `app` layer to the exceptions and it can have an index file, should I handle this edge case when a lower layer tries to import from app/index?
